File: 프로그래머스/unrated/160586. 대충 만든 자판/대충 만든 자판.py
def solution(keymap, targets):
    dic = {}
    for k in keymap:
        for idx, value in enumerate(k):
            if value not in dic:
                dic[value] = idx+1  
            elif dic[value] > idx+1:
                dic[value] = idx+1
    
    result = []
    
    for tar in targets:
        tmp = 0
        flag = False
        for i in range(len(tar)):
            if tar[i] not in dic:
                flag = True
                break
            else:
                tmp += dic[tar[i]]
        if flag:
            result.append(-1)
        else:
            result.append(tmp)
    return result
                
        

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int[] solution(String[] keymap, String[] targets) {
        Map<Character, Integer> dic = new HashMap<>();
        for (int i = 0; i < keymap.length; i++) {
            for (int j = 0; j < keymap[i].length(); j++) {
                char c = keymap[i].charAt(j);
                if (dic.containsKey(c)) {
                    if (dic.get(c) > j + 1) {
                        dic.put(c, j + 1);
                    }
                } else {
                    dic.put(c, j + 1);
                }
            }
        }
        
        int[] result = new int[targets.length];
        
        for (int i = 0; i < targets.length; i++) {
            int answer = 0;
            char[] targetArr = targets[i].toCharArray();
            for (char c : targetArr) {
                if (dic.containsKey(c)) {
                    answer += dic.get(c);
                } else {
                    answer = -1;
                    break;
                }
            }
            
            result[i] = answer;
        }
        
        return result;
    }
}